dataland-frontend/src/components/pages/CompanyInformation.vue

```diff
@@ -120,15 +126,36 @@ export default defineComponent({
   mounted() {
     void this.getCompanyInformation();
     void this.setDataOwnershipStatus();
+    void this.updateHasCompanyDataOwner();
   },
   watch: {
-    companyId() {
-      void this.getCompanyInformation();
-      void this.setDataOwnershipStatus();
-      this.claimIsSubmitted = false;
+    async companyId(newCompanyId, oldCompanyId) {
+      if (newCompanyId !== oldCompanyId) {
+        try {
+          void this.setDataOwnershipStatus();
+
+          void this.getCompanyInformation();
+          this.hasCompanyDataOwner = await getCompanyDataOwnerInformation(
+            assertDefined(this.getKeycloakPromise),
+            newCompanyId as string,
+          );
+          this.claimIsSubmitted = false;
+        } catch (error) {
+          console.error("Error fetching data for new company:", error);
+        }
+      }
     },
   },
   methods: {
+    /**
+     * updates the hasCompanyDataOwner in an async way
```

Comment:
Updates

Reply:
Ist bei mir noch klein geschrieben. Kannst du einmal in der Datei hier durchgehen und die Docs mit Großbuchstaben beginnen lassen.

Reply:
Sorry, das ist mir durchgerutscht.

Reply:
Top, Danke dir